esmvalcore/_task.py

```diff
@@ -301,10 +301,10 @@ def _initialize_cmd(self):
                 'ncl': ['-n', '-p'],
             }
             if self.settings['profile_diagnostic']:
-                profile_file = Path(self.settings['run_dir']) / 'profile.bin'
+                profile_file = Path(self.settings['run_dir']) / 'profile.json'
                 args['py'] = [
-                    '-m', 'vmprof', '--lines', '-o',
-                    str(profile_file)
+                    '-m', 'vprof', '-o',
+                    str(profile_file), '-c', 'p'
```

Comment:
```suggestion
                    str(profile_file), '-c', 'c'
```
Using the statistical profiler probably causes the lowest overhead and gives the most accurate results.

Reply:
I am going to try it right now, I found out that using multiple options eg `-c cmhp` makes the diag crash with `FileExistsError: [Errno 17] File exists: '/home/users/valeriu/esmvaltool_var_test/Ophelie/recipe_python_20200915_151109/work/diagnostic1/script1'` - it looks like it's running multiple instances of the same diag - if you use just `p` all's good; trying `c` now

Reply:
note that `c` is CPU flame graph - is that what you're after? I don't know what that means :grin:

Reply:
OK flame graph produces an error:
```
2020-09-15 15:24:21,258 [3936] INFO     esmvaltool.diag_scripts.shared._base,480        End of diagnostic script run.
Traceback (most recent call last):
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/__main__.py", line 102, in <module>
    main()
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/__main__.py", line 83, in main
    program_stats = runner.run_profilers(source, config, verbose=True)
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/runner.py", line 73, in run_profilers
    run_stats[option] = curr_profiler.run()
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/base_profiler.py", line 172, in run
    return self.profile()
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/flame_graph.py", line 167, in profile_module
    return base_profiler.run_in_separate_process(self._profile_module)
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/base_profiler.py", line 79, in run_in_separate_process
    raise exc
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/base_profiler.py", line 49, in run
    self._target(*self._args, **self._kwargs))
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/flame_graph.py", line 155, in _profile_module
    call_tree = prof.call_tree
  File "/home/users/valeriu/anaconda3R/envs/esmvaltool/lib/python3.8/site-packages/vprof/flame_graph.py", line 107, in call_tree
    for stack, sample_count in self._stats.items():
RuntimeError: dictionary changed size during iteration
```

Reply:
Yes, that's what we want, here's an introduction from the [inventor](https://queue.acm.org/detail.cfm?id=2927301). Basically a statistical profiler looks at the call stack e.g. every millisecond and checks what function is executing. It just counts how many times it sees a particular function. Thus if a function has a high count, you know that your program is spending most of it's time in that function, without having to time every single function called. That makes it much lower overhead than a classical profiler. A flame graph is a nice visualization of that result, just check it out by running the `vprof --input-file /path/to/profile.json` command.

Reply:
>  looks like it's running multiple instances of the same diag

Indeed, just use a single profiler, I think `c` is a good default

Reply:
can't run with `-c c` coz of that error :arrow_up:

Reply:
OK ran it on the new Jasmin sci2 node and it works well with `-c c` (it was failing on an old sci5 that has now been decommissioned)